mesonbuild/modules/gnome.py

```diff
@@ -479,8 +479,11 @@ def compile_resources(self, state: 'ModuleState', args: T.Tuple[str, 'FileOrStri
             else:
                 raise MesonException('Compiling GResources into code is only supported in C and C++ projects')
 
-        if kwargs['install'] and not gresource:
-            raise MesonException('The install kwarg only applies to gresource bundles, see install_header')
+        if kwargs['install']:
+            if not gresource:
+                raise MesonException('The install kwarg only applies to gresource bundles, see install_header')
+            elif not kwargs['install_dir']:
+                raise MesonException('The install_dir kwarg is required for gresource bundles to install')
```

Comment:
We have a consistent way to describe this message with other functions that need the same thing. e.g.

```python
raise InvalidArguments('i18n.merge_file: "install_dir" keyword argument must be set when "install" is true.')
```

The issue is about kwargs that always need to be used together, which is a generic enough problem that getting into details about gresources isn't necessary.

Reply:
alright